src/esp/bindings_js/modules/vr_demo.js

```diff
@@ -25,96 +28,259 @@ class VRDemo extends WebDemo {
+        this.gl.TEXTURE_WRAP_T,
+        this.gl.CLAMP_TO_EDGE
+      );
+      this.gl.texParameteri(
+        this.gl.TEXTURE_2D,
+        this.gl.TEXTURE_MIN_FILTER,
+        this.gl.NEAREST
+      );
+      this.gl.texParameteri(
+        this.gl.TEXTURE_2D,
+        this.gl.TEXTURE_MAG_FILTER,
+        this.gl.NEAREST
+      );
+    }
 
-  drawVRScene() {
-    this.vrSceneFrame = this.currentVrDisplay.requestAnimationFrame(
-      this.drawVRScene.bind(this)
+    this.gl.bindTexture(this.gl.TEXTURE_2D, this.tex);
+    this.gl.texImage2D(
+      this.gl.TEXTURE_2D,
+      0, // level
+      this.gl.RGBA, // internal format
+      texRes[1], // width
+      texRes[0], // height
+      0, // border
+      this.gl.RGBA, // format
+      this.gl.UNSIGNED_BYTE, // type
+      new Uint8Array(texData) // data
+    );
+
+    this.gl.activeTexture(this.gl.TEXTURE0);
+
+    this.gl.drawElements(
+      this.gl.TRIANGLES,
+      this.inds.length,
+      this.gl.UNSIGNED_SHORT,
+      0
     );
-    this.currentVrDisplay.getFrameData(this.frameData);
-    this.updateAgentState();
-    this.task.render({ renderTopDown: false });
-    this.currentVrDisplay.submitFrame();
-    this.updateFPS();
   }
 
-  updateAgentState() {
+  drawVRScene(t, frame) {
+    let session = frame.session;
+
+    session.requestAnimationFrame(this.drawVRScene.bind(this));
+
+    let pose = frame.getViewerPose(this.xrReferenceSpace);
+
+    // Need the pose to render a frame
+    if (!pose) {
+      return;
+    }
+
+    let layer = session.renderState.baseLayer;
+    this.gl.bindFramebuffer(this.gl.FRAMEBUFFER, layer.framebuffer);
+
     const agent = this.simenv.sim.getAgent(this.simenv.selectedAgentId);
-    let prevAgentState = this.simenv.createAgentState({});
-    agent.getState(prevAgentState);
 
-    const simRotation = prevAgentState.rotation.slice();
-    const simPosition = prevAgentState.position.slice();
+    const VIEW_SENSORS = ["left_eye", "right_eye"];
+    for (var iView = 0; iView < pose.views.length; ++iView) {
+      let view = pose.views[iView];
+      let viewport = layer.getViewport(view);
+      this.gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
 
-    if (!this.previousRotation) {
-      this.previousRotation = this.frameData.pose.orientation.slice();
-      this.previousPosition = this.frameData.pose.position.slice();
-    } else {
-      const poseRotation = this.frameData.pose.orientation;
-      const posePosition = this.frameData.pose.position;
+      let sensor = agent.sensorSuite.get(VIEW_SENSORS[iView]);
 
-      poseRotation.forEach((item, index) => {
-        simRotation[index] += item - this.previousRotation[index];
-      });
-      posePosition.forEach((item, index) => {
-        simPosition[index] += item - this.previousPosition[index];
-      });
-      this.previousRotation = poseRotation.slice();
-      this.previousPosition = posePosition.slice();
+      const pointToArray = p => [p.x, p.y, p.z, p.w];
```

Comment:
Does this lambda need to be regenerated twi every frame. Might want to move it out of this hot code.